src/big/big_float.cr

```diff
@@ -149,6 +149,36 @@ struct BigFloat < Float
   Number.expand_div [BigDecimal], BigDecimal
   Number.expand_div [BigRational], BigRational
 
+  def **(other : BigInt) : BigFloat
+    is_zero = self.zero?
+    if is_zero
+      case other <=> 0
+      when .>(0)
+        return self
+      when .<(0)
```

Comment:
I'm wondering: Is the duplicate comparison on purpose in order to call `BigInt#<=>` only once? Then I'd suggest to mention that in a comment to avoid future refactorings getting rid of it accidentally.
Otherwise, this could directly be written a bit simpler as:
```suggestion
      case other
      when .>(0)
        return self
      when .<(0)
```